editor_canvas.py

```
@@ -82,53 +148,137 @@ def find_class_at(self, x, y) -> UMLClass | None:
+        if hits:
+            # Return the closest one
+            hits.sort(key=lambda item: item[0])
+            return hits[0][1]
+            
+        return None
+
     def on_button_press(self, event):
         if self.editor_widget:
-            if not self.commit_edit():
-                return # Still editing due to validation error, don't select or drag
+            if not self.commit_edit(): return
```

Ruff (0.15.6)

[warning] Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)